src/apps/explorer/pages/Metadata/config.ts

```diff
@@ -0,0 +1,144 @@
+    metadata: {
+      type: 'object',
+      title: 'Metadata',
+      properties: {
+        referrer: {
+          type: 'object',
+          title: 'Referrer',
+          properties: {
+            enableReferrer: { type: 'boolean', title: 'Enable/Disable' },
+          },
+          dependencies: {
+            enableReferrer: {
+              oneOf: [
+                {
+                  properties: {
+                    enableReferrer: {
+                      const: false,
+                    },
+                  },
+                },
+                {
+                  properties: {
+                    enableReferrer: {
+                      const: true,
+                    },
+                    address: { type: 'string', title: 'Address' },
+                    version: {
+                      type: 'string',
+                      title: 'Version',
+                      readOnly: true,
+                      default: LATEST_REFERRER_METADATA_VERSION,
+                    },
+                  },
+                  required: ['version'],
+                },
+              ],
+            },
+          },
+        },
+        quote: {
+          type: 'object',
+          title: 'Quote',
+          properties: {
+            enableQuote: { type: 'boolean', title: 'Enable/Disable' },
+          },
+          dependencies: {
+            enableQuote: {
+              oneOf: [
+                {
+                  properties: {
+                    enableQuote: {
+                      const: false,
+                    },
+                  },
+                },
+                {
+                  properties: {
+                    enableQuote: {
+                      const: true,
+                    },
+                    version: {
+                      type: 'string',
+                      title: 'Version',
+                      default: LATEST_QUOTE_METADATA_VERSION,
+                      readOnly: true,
+                    },
+                    slippageBips: { type: 'string', title: 'Slippage Bips', pattern: '^\\d*$' },
+                  },
+                  required: ['version'],
+                },
+              ],
+            },
+          },
+        },
+      },
+    },
+  },
+}
+
+export const getSchema = async (): Promise<JSONSchema7> => {
+  const { default: latestSchema } = await import(`@cowprotocol/app-data/schemas/v${LATEST_APP_DATA_VERSION}.json`)
+  return { ...latestSchema, ...schema } as JSONSchema7
+}
+
+export const uiSchema: UiSchema = {
+  environment: {
+    'ui:help': 'Hint: Development, Production, Staging.',
+  },
+}
+
+export const validate = (formData: any, errors: FormValidation): FormValidation => {
+  const { address, enableReferrer } = formData.metadata.referrer
+  if (enableReferrer && !isAddress(address?.toLowerCase())) {
+    errors.metadata['referrer'].address.addError('This is not an address')
+  }
+  return errors
```

Comment:
why not leverage the `pattern` field the current json schema has? The lib uses ajv for validation so it works the same as the validation we do in the sdk:

```json
"address": {
  "title": "Referrer address",
  "$id": "#/definitions/ethereumAddress",
  "pattern": "^0x[a-fA-F0-9]{40}$",
  "examples": [
    "0xb6BAd41ae76A11D10f7b0E664C5007b908bC77C9"
  ],
  "type": "string"
}
```

docs: https://react-jsonschema-form.readthedocs.io/en/latest/usage/validation/